iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp

```diff
@@ -256,13 +256,15 @@ scf::ParallelOp collapseParallelLoops(ConversionPatternRewriter &rewriter,
   rewriter.setInsertionPointToStart(&newPLoopOp.getLoopBody().front());
   Value loopIv = *newPLoopOp.getInductionVars().begin();
   BlockAndValueMapping map;
+  edsc::ScopedContext scope(rewriter, loc);
+  using namespace edsc::op;
+  Value two = rewriter.create<ConstantIndexOp>(loc, 2);
   for (int i : llvm::seq<int>(0, numLoops)) {
     Value iterNum =
         rewriter.create<SignedDivIOp>(loc, loopIv, iterationStride[i]);
-    Value newIv = rewriter.create<AddIOp>(
-        loc, lbs[i], rewriter.create<MulIOp>(loc, iterNum, steps[i]));
+    Value newIv = lbs[i] + (iterNum * steps[i]);
     map.map(pLoopBody.getArgument(i), newIv);
-    loopIv = rewriter.create<SignedRemIOp>(loc, loopIv, iterationStride[i]);
+    if (i != numLoops - 1) loopIv = loopIv % two;
```

Comment:
I am not sure why this is `% 2`

Reply:
That was a bad hack, thankfully it got caught by some tests. I reverted this line to not use affine expression as it is not valid in case the divisor is not constant and I don't really need it right now.